### PR TITLE
📈 Add Marketing Components

### LINKED ISSUE #20: Analytics - add marketing components
- [x] Google Tag Manager
- [x] Google Analytics
- [x] Hotjar
- [x] Zendesk

### PR DESCRIPTION
**Pull request by tkapa**
Closes #20 

Implements Google Analytics and Google Tag Manager using GetAnalytics.io.

**Note:** We may be able to ask GetAnalytics to implement Hotjar and Zendesk, that way we can manage analytics in one place. 

GA and GTM will not be configured if the Secrets are empty. Hotjar and Zendesk will not be initialized while in development mode.

Implements Hotjar using react-hotjar.

No Zendesk Packages stood out. Implemented Chat using Next/Script.

**Note:** It's possible that using a [custom document](https://nextjs.org/docs/advanced-features/custom-document) might allow us to improve implementation

TODO before merging: 
- [x] Add Secrets to the GitHub repo using service accounts (GA, GTM, Hotjar, Zendesk)
- [x] Note the PageSpeed scores before and after adding the Analytics

### CODE REVIEW

**Comment by tkapa** [comment]
The deployment doesn't seem to show the analytics components... Looking into the fix.

**Comment by wicksipedia** [comment]
> The deployment doesn't seem to show the analytics components... Looking into the fix.

Looks like you haven't added them to the `Dockerfile`

**Comment by tkapa** [comment]
Updated the Actions appropriately

Can now see Hotjar, GA, GTM and Zendesk active on the website.
image

**Comment by tkapa** [comment]
Even after attempting to implement NextJs/Script features, it seems that PageSpeed insights isn't getting much better.

We might just be better off taking the Performance hit to get the features in. 

- [ ] ⚡ PageSpeed Insights | Address Performance Issues

**Comment by tkapa** [comment]
Hey @wicksipedia,

Zendesk seems to be the biggest culprit of the long page load times. Given that the PageSpeed Insights we receive are based on a mobile with slow 4G, should we look to disabling Zendesk on mobile?

**Comment by wicksipedia** [comment]
@camillars - do we need zendesk on mobile?

**Comment by camillars** [comment]
> @camillars - see above... do we need zendesk on mobile?

Hi team @wicksipedia @tkapa 

Ideally, we should have the same experience on mobile since sometimes we sell tickets via mobile on Eventbrite.

Would we have an option to at least allow users to submit a ticket with a question?

**Comment by tkapa** [comment]
> Would we have an option to at least allow users to submit a ticket with a question?  

It doesn't seem like we have that kind of flexibility with Zendesk, unless there are some options wherever the widget is configured to allow for that kind of thing.

**Comment by wicksipedia** [comment]
@tkapa does this help?
https://support.zendesk.com/hc/en-us/articles/4408835776794-The-Zendesk-Widget-is-slowing-down-my-website-speed-and-load-time

**Comment by tkapa** [comment]
> @tkapa does this help? [support.zendesk.com/hc/en-us/articles/4408835776794-The-Zendesk-Widget-is-slowing-down-my-website-speed-and-load-time](https://support.zendesk.com/hc/en-us/articles/4408835776794-The-Zendesk-Widget-is-slowing-down-my-website-speed-and-load-time)

From what I know I don't believe that we're using the Classic widget. But I'll see if the same principals could apply.

**Comment by tkapa** [comment]
> @tkapa does this help? [support.zendesk.com/hc/en-us/articles/4408835776794-The-Zendesk-Widget-is-slowing-down-my-website-speed-and-load-time](https://support.zendesk.com/hc/en-us/articles/4408835776794-The-Zendesk-Widget-is-slowing-down-my-website-speed-and-load-time)

Ironically a further article in this says that loads within 2-3 seconds are acceptable for performance.

https://support.zendesk.com/hc/en-us/articles/4408834314266-What-s-considered-a-good-loading-time-for-the-Chat-widget-

Happy to trust your judgement on whether we are OK with taking this hit to perhaps address the issues on PageSpeed Insights at a later time. (In #59 )

**Comment by wicksipedia** [comment]
https://www.newmediacampaigns.com/blog/maintaining-great-site-performanc-using-zendesk-web-widget

File: components/layout/analytics.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import Script from "next/script";
+
+const zendeskKey = process.env.NEXT_PUBLIC_ZENDESK_CHAT_KEY;
+const gaId = process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS;
+const gtmId = process.env.NEXT_PUBLIC_GOOGLE_GTM_ID;
+const hjId = process.env.NEXT_PUBLIC_HOTJAR_ID;
+const hjSv = process.env.NEXT_PUBLIC_HOTJAR_SV;
+
+export const Analytics = () => {
```
Comment (wicksipedia) [inline components/layout/analytics.tsx:9]: can we separate the different analytics bits into separate components (in this file)?

Should make it easier to read that a big wall of script tags

**Review by wicksipedia — CHANGES_REQUESTED** [review]
lgtm otherwise

**Comment by wicksipedia** [comment]
> Happy to trust your judgement on whether we are OK with taking this hit to perhaps address the issues on PageSpeed Insights at a later time. (In https://github.com/SSWConsulting/SSW.Website-v3/issues/59 )

happy to respect the timebox - i'll add perf comments to the other issue

**Comment by JackDevAU** [comment]
I wonder if using the lazy loading and suspense in React 18 + moving to NextJS v13 would help with this issue. I think it would be worth it to investigate. I know these are still "experimental" but these are super handy features.

Checkout:
https://nextjs.org/docs/advanced-features/dynamic-import
https://nextjs.org/docs/advanced-features/react-18/overview

**Comment by wicksipedia** [comment]
Good call @JackDevAU - best to put that on #59
ATM we are on Next 12/React 17 but can look to upgrade

File: components/layout/analytics.tsx (hunk 1 above)
Comment (tkapa) [reply components/layout/analytics.tsx:9]: Done - Looks a bit better now

**Review by tkapa — COMMENTED** [review]
Done - Looks a bit better now

**Review by wicksipedia — APPROVED** [review]

**Pull request merged**